File: python/search/linear.py
from collections.abc import Sequence


def linear_search[T](values: Sequence[T], target: T) -> int:
    return values.index(target)


def linear_search_manual[T](values: Sequence[T], target: T) -> int:
    for i, value in enumerate(values):
        if value is target or value == target:
            return i
    raise ValueError("linear_search_manual() target not in sequence")

File: go/search/linear.go
package search

import "slices"

func LinearSearch[S ~[]E, E comparable](s S, target E) int {
	return slices.Index(s, target)
}

func LinearSearchManual[S ~[]E, E comparable](s S, target E) int {
	for i := range s {
		if s[i] == target {
			return i
		}
	}
	return -1
}
